File: medium/bob.js
const isEmpty = (message) => !/\S/.test(message);
const isQuestion = (message) => /\?\s*$/.test(message);
const isShout = (message) => /^[^a-z]*[A-Z][^a-z]*$/.test(message);

export const hey = (message) => {
  if (isQuestion(message)) {
    return isShout(message) ? "Calm down, I know what I'm doing!" : "Sure.";
  }

  if (isShout(message)) {
    return "Whoa, chill out!";
  }

  return isEmpty(message) ? "Fine. Be that way!" : "Whatever.";
};

File: python/bob.py
def response(hey_bob):
    formatted = hey_bob.strip(" \n\r\t")
    if formatted == "":
        return "Fine. Be that way!"
    elif formatted[-1] == "?":
        if formatted.isupper():
            return "Calm down, I know what I'm doing!"
        return "Sure."
    elif formatted.isupper():
        return "Whoa, chill out!"
    else:
        return "Whatever."
